integration_tests/src/main/python/subquery_test.py

```diff
@@ -13,122 +13,185 @@
 # limitations under the License.
 
 import pytest
-from asserts import assert_gpu_and_cpu_are_equal_sql
+from asserts import assert_gpu_and_cpu_are_equal_sql, assert_gpu_and_cpu_error
+from conftest import spark_tmp_table_factory
 from data_gen import *
 from marks import *
 
 @ignore_order(local=True)
 @pytest.mark.parametrize('data_gen', all_basic_gens, ids=idfn)
-def test_scalar_subquery_basics(data_gen):
+def test_scalar_subquery_basics(spark_tmp_table_factory, data_gen):
     # Fix num_slices at 1 to make sure that first/last returns same results under CPU and GPU.
+    table_name = spark_tmp_table_factory.get()
     assert_gpu_and_cpu_are_equal_sql(
         lambda spark: gen_df(spark, [('a', data_gen)], num_slices=1),
-        'table',
-        '''select a, (select last(a) from table)
-        from table
-        where a > (select first(a) from table)
-        ''')
+        table_name,
+        '''select a, (select last(a) from {})
+        from {}
+        where a > (select first(a) from {})
+        '''.format(table_name, table_name, table_name))
```

Comment:
Nit: This is more readable and less error-prone with Python's f-strings:
```suggestion
        f'''select a, (select last(a) from {table_name})
        from {table_name}
        where a > (select first(a) from {table_name})'''
```
Comment applies to the other places where `.format` is used.

Reply:
Agreed. I've fixed this now.